### PR TITLE
Added HTTPS

### PR DESCRIPTION
**Pull request by AToska21**

### CODE REVIEW

**Comment by AToska21** [comment]
@ChrisChrome

**Comment by TheArcaneBrony** [comment]
Looks good as far as I can see, are you sure adding the one with certbot's configuration is a good idea? I feel like it would prompt people to copy this and have a nonfunctional setup.

**Comment by AToska21** [comment]
Yes, it was recommended to me by ChrisChrome. That's what I use. It works perfectly.

**Comment by AToska21** [comment]
There. should be better.

**Review by TheArcaneBrony — APPROVED** [review]
Looking better, this should make it clear enough. Thanks!

**Comment by AToska21** [comment]
> Looking better, this should make it clear enough. Thanks!

No problem. I was having problems figuring this out. Thanks ChrisChrome!

**Review by omxpro — COMMENTED** [review]
This should be called, how to reverse proxy your domain and add ssl to it

**Comment by TheArcaneBrony** [comment]
@AToska21 you shouldnt change the build.yml.
@omxpro I feel like this would be too technical in terms of wording. Keep in mind not everye knows what SSL or reverse proxying is.

**Comment by AToska21** [comment]
i know i shouldn't change it i was dumb lol

**Comment by omxpro** [comment]
> @AToska21 you shouldnt change the build.yml. @omxpro I feel like this would be too technical in terms of wording. Keep in mind not everye knows what SSL or reverse proxying is.

Hmmm tbf yeaa

**Comment by omxpro** [comment]
Sorry

**Comment by TheArcaneBrony** [comment]
No worries, u good

**Comment by AToska21** [comment]
@ChrisChrome ?

File: docs/setup/HTTPS.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+
+### Using HTTPS
+
+Using the existing HTTP config, run this command
+
+Generate your certificates with certbot.
+
+``sudo certbot --nginx -d fosscord.your.website``
+
```
Comment (ChrisChrome) [inline docs/setup/HTTPS.md:67]: Add instructions for automatic renewal via crontab or systemd timers

File: docs/setup/HTTPS.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                proxy_set_header  X-Forwarded-Proto https;
+                proxy_set_header  X-Forwarded-For $remote_addr;
+                proxy_set_header  X-Forwarded-Host $remote_addr;
+                proxy_no_cache 1;
+                proxy_cache_bypass 1;
+                proxy_set_header Upgrade $http_upgrade;
+                proxy_set_header Connection "upgrade";
+    }
+
+        location = /api/v9/auth/register {
```
Comment (ChrisChrome) [inline docs/setup/HTTPS.md:39]: Place a comment above this block noting that this is what you would remove to disable rate limiting

**Review by ChrisChrome — CHANGES_REQUESTED** [review]

File: docs/setup/HTTPS.md (hunk 1 above)
Comment (AToska21) [reply docs/setup/HTTPS.md:67]: how?

Comment (ChrisChrome) [reply docs/setup/HTTPS.md:68]: I'll look for how to do systemd timers, but crontab is simple enough
`0 12 * * * /usr/bin/certbot renew --quiet` in the crontab file, which can be edited with `crontab -e` on any system with crontab

Comment (AToska21) [reply docs/setup/HTTPS.md:68]: done!

**Comment by AToska21** [comment]
I've finished with the edits. This pull request is ready to merge.

**Review by ChrisChrome — APPROVED** [review]

File: docs/setup/HTTPS.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+A reverse proxy is when you take a web server, for our example, NGINX and you ue it to redirect traffic on port 80/443 to port 3001, but to
+your browser, it still appears as port 80/443
+
+A reverse proxy is the most efficient way to setup HTTPS on your instance.
+
+To do this, you need a basic understanding of setting up a web server.
+
+### Without HTTPS (you can remove the "location = api/register" part of the website, it's there to stop people from raiding your instance)
+
+
```
Comment (erkinalp) [inline docs/setup/HTTPS.md:16]:
```suggestion
```

File: docs/setup/HTTPS.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        server_name fosscord.your.website;
+        listen 80;
+    return 404; # managed by Certbot
+
+
+}
+```
+
```
Comment (erkinalp) [inline docs/setup/HTTPS.md:136]:
```suggestion
```

**Review by erkinalp — CHANGES_REQUESTED** [review]

**Review by erkinalp — APPROVED** [review]

**Pull request merged**